api/python/quilt3/data_transfer.py

```diff
@@ -429,6 +426,38 @@ def __init__(self, s3_client_provider, progress, done, run):
         self.run = run
 
 
+def _get_executor(maximum_concurrency: int = None):
+    # Dask clients and concurrent futures executors have the same API of submit and map
+    # (Specifically designed to be that way)
+    # As more and more people are starting to use dask in data engineering and modeling
+    # workloads, the problem that arises is it is common for single dask workers to
+    # only have a few cores / threads.
+    # Instead of utilizing a single workers available resources we can call out to
+    # the scheduler that we want to distribute our jobs to the available cluster.
+    # https://docs.dask.org/en/latest/futures.html#start-dask-client
+
+    # Check if this function is running in a dask worker
+    try:
```

Comment:
Not sure if picking up dask automatically is a good idea. Probably safer to use a config option?

Also, what about unit tests?

Reply:
I couldn't get the unit tests working with this change which is why I marked it as a WIP PR. as soon as I get them working for the base case of threadpoolexecutor I will add tests for dask worker case.

What about an env variable?
`QUILT_USE_DASK=True`?